### PR TITLE
delete old applied work after its new work is applied

### PR DESCRIPTION
**Pull request by skeeey**
Signed-off-by: Wei Liu <liuweixa@redhat.com>

### CODE REVIEW

**Review by qiujian16 — COMMENTED** [review]
/approve
/lgtm

File: pkg/spoke/controllers/finalizercontroller/unmanaged_appliedmanifestwork_controller.go [hunk 1]
```diff
@@ -71,6 +79,26 @@ func (m *UnManagedAppliedWorkController) sync(ctx context.Context, controllerCon
 		// work not found, could have been deleted, do nothing.
 		return nil
 	}
+	if err != nil {
+		return err
+	}
+
+	// We delete the old applied work until the work of this applied work is applied.
```
Comment (BenamarMk) [inline pkg/spoke/controllers/finalizercontroller/unmanaged_appliedmanifestwork_controller.go:86]: Do you mean `We don't delete` instead of `We delete`. Let me rephrase it to see if I can understand it:
```
We don't delete the old AppliedManifestWork until the new one is applied
```
Also, where is the delete???

Comment (elgnay) [reply pkg/spoke/controllers/finalizercontroller/unmanaged_appliedmanifestwork_controller.go:86]: > Do you mean `We don't delete` instead of `We delete`. Let me rephrase it to see if I can understand it:
> 
> ```
> We don't delete the old AppliedManifestWork until the new one is applied
> ```

Yes, you are right. There is a typo in the comment. It should be
```
We don't delete the old AppliedManifestWork until the new one is applied
```

> 
> Also, where is the delete???

The delete logic can be found here:
https://github.com/open-cluster-management-io/work/blob/main/pkg/spoke/controllers/finalizercontroller/unmanaged_appliedmanifestwork_controller.go#L109-L115

Comment (skeeey) [reply pkg/spoke/controllers/finalizercontroller/unmanaged_appliedmanifestwork_controller.go:86]: yes, this pr postpones the deletion operation,  we need wait the new appliedmanifest applied to avoid deleting the old applied work prematurely
